okhttp/src/main/java/okhttp3/android/OkHttpAndroidClientFactory.kt

```diff
@@ -0,0 +1,70 @@
+package okhttp3.android
+
+import android.content.Context
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import okhttp3.internal.platform.Platform
+import okhttp3.internal.platform.android.BuildConfig
+import okhttp3.internal.tls.AllowlistedTrustManager
+
+class OkHttpAndroidClientFactory private constructor(val client: OkHttpClient) {
+    fun newClient(init: OkHttpClient.Builder.() -> Unit): OkHttpClient {
+        val builder = client.newBuilder()
+        init(builder)
+        return builder.build()
+    }
+
+    fun shutdown() {
```

Comment:
FWIW part of the reason for shutdown here, is that if you have a Cache you probably also want to close it, since that does some cleanup.  And this method could be hooked int App Lifecycle.